lib/active_merchant/billing/gateways/credorax.rb

```diff
@@ -283,19 +283,22 @@ def add_invoice(post, money, options)
         'maestro' => '9'
       }
 
-      def add_payment_method(post, payment_method)
+      def add_payment_method(post, payment_method, options)
         post[:c1] = payment_method&.name || ''
-        add_network_tokenization_card(post, payment_method) if payment_method.is_a? NetworkTokenizationCreditCard
+        add_network_tokenization_card(post, payment_method, options) if payment_method.is_a? NetworkTokenizationCreditCard
         post[:b2] = CARD_TYPES[payment_method.brand] || ''
         post[:b1] = payment_method.number
         post[:b5] = payment_method.verification_value
         post[:b4] = format(payment_method.year, :two_digits)
         post[:b3] = format(payment_method.month, :two_digits)
       end
 
-      def add_network_tokenization_card(post, payment_method)
+      def add_network_tokenization_card(post, payment_method, options)
         post[:b21] = NETWORK_TOKENIZATION_CARD_SOURCE[payment_method.source.to_s]
-        post[:token_eci] = payment_method&.eci if payment_method.source.to_s == 'network_token'
+        post[:token_eci] = '07'
+        post[:token_eci] = '00' if payment_method.brand.to_s == 'master'
+        post[:token_eci] =  options[:eci] if options[:eci]
+        post[:token_eci] =  options[:eci] || payment_method&.eci if payment_method.source.to_s == 'network_token'
```

Comment:
💬 Comment:

I get the point of the code, but we need a simpler logic, something like:

```Ruby
 post[:token_eci] =  options[:eci] || payment_method&.eci || payment_method.brand.to_s == 'master' ? '00' : '07'
# Note: you are using the safe operator so don't need to check for payment method source to try to get the eci
```


```